src/stdatamodels/jwst/datamodels/schemas/specwcs_nircam_grism.schema.yaml

```
@@ -15,37 +15,49 @@ allOf:
           Nircam Grism wavelength dispersion model, instance of astropy.modeling.Model
         type: array
         items:
-          type: object
+          type: array
+          items:
+            type: object
```

This will likely be incompatible with existing files as they will contain a single instance of the object rather than an array containing a single object. This may cause a validation error when reading such files.

If there is a validation error with existing files then
```suggestion
          oneOf:
            - type: object
            - type: array
               items:
                 type: object
```
Should resolve that error. This will apply to all the similar changes in this file.

---

When written (a few months ago), there was no intent to preserve backwards compatibility as the prior versions were barely "science-valid". I don't know if this stance has changed, but other parts of the code are also implicitly backwards incompatible and would break if used with older references. Will be something to discuss once this is closer to review.